core/credential/credential.go

```diff
@@ -0,0 +1,44 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package credential
+
+import "context"
+
+type Credentials struct {
+	Host     string
+	Username string
+	Secret   string
+}
+
+// Manager is Credential manager plugin core method
```

Comment:
The storing should be optional and we can probably skip it in the initial versions.

Reply:
the current user can modify the `config.toml` file to specify credential storing type, and the user not specify has a default value.
```toml
[plugins.'io.containerd.credential.v1.credential']
    creds_store = 'file'
```
default value rule:
By default, containerd looks for the native binary on each of the platforms, i.e. "osxkeychain" on macOS, "wincred" on windows, and "pass" on Linux. A special case is that on Linux, containerd will fall back to the "secretservice" binary if it cannot find the "pass" binary. If none of these binaries are present, it stores the credentials (i.e. password) in base64 encoding in the config files described above.